src/virtio/block.c

```diff
@@ -0,0 +1,493 @@
+#include <microkit.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include "virtio/config.h"
+#include "virtio/virtq.h"
+#include "virtio/mmio.h"
+#include "virtio/block.h"
+#include "virq.h"
+#include "util.h"
+#include <sddf/blk/fsmalloc.h>
+#include <sddf/util/ialloc.h>
+#include <sddf/blk/queue.h>
+
+/* Uncomment this to enable debug logging */
+// #define DEBUG_BLOCK
+
+#if defined(DEBUG_BLOCK)
+#define LOG_BLOCK(...) do{ printf("VIRTIO(BLOCK): "); printf(__VA_ARGS__); }while(0)
+#else
+#define LOG_BLOCK(...) do{}while(0)
+#endif
+
+#define LOG_BLOCK_ERR(...) do{ printf("VIRTIO(BLOCK)|ERROR: "); printf(__VA_ARGS__); }while(0)
+
+/* Maximum number of buffers in sddf data region */
+//@ericc: as long as this number is higher than any sddf_handler->data_size we should be fine
```

Comment:
Turn this into a runtime assertion in `virtio_blk_init` and write a comment there explaining the reasoning. I think that should be sufficient.